core/runtime/TRTEngine.cpp

```diff
@@ -21,14 +24,35 @@ TRTEngine::TRTEngine(std::string serialized_engine)
           util::logging::get_logger().get_reportable_severity(),
           util::logging::get_logger().get_is_colored_output_on()) {
   std::string _name = "deserialized_trt";
-  new (this) TRTEngine(_name, serialized_engine);
+  new (this) TRTEngine(_name, serialized_engine, empty_string);
 }
 
-TRTEngine::TRTEngine(std::string mod_name, std::string serialized_engine)
+TRTEngine::TRTEngine(std::vector<std::string> serialized_info)
+    : logger(std::string("[] = "),
+        util::logging::get_logger().get_reportable_severity(),
+	util::logging::get_logger().get_is_colored_output_on()) {
+    std::string _name = "deserialized_trt";
+    std::string device_info = serialized_info[0];
+    std::string engine_info = serialized_info[1];
+
+    new (this) TRTEngine(_name, engine_info, device_info);
+}
+
+TRTEngine::TRTEngine(std::string mod_name, std::string serialized_engine,
+		std::string serialized_device_info = empty_string)
     : logger(
           std::string("[") + mod_name + std::string("_engine] - "),
           util::logging::get_logger().get_reportable_severity(),
           util::logging::get_logger().get_is_colored_output_on()) {
+  CudaDevice cuda_device;
+  // Deserialize device meta data if device_info is non-empty
+  if (!serialized_device_info.empty())
+  {
+      cuda_device = deserialize_device(serialized_device_info);
+      // Set CUDA device as configured in serialized meta data
+      set_cuda_device(cuda_device);
```

Comment:
do we store the deserialized device somewhere?

Reply:
We don't save the deserialized device information anywhere. Do you want to perform file write in runtime?